File: gateway/src/main/java/com/beat/gateway/security/internal/servlet/AdminAuthentication.java
package com.beat.gateway.security.internal.servlet;

import java.util.Collection;
import org.springframework.security.authentication.UsernamePasswordAuthenticationToken;
import org.springframework.security.core.GrantedAuthority;

public class AdminAuthentication extends UsernamePasswordAuthenticationToken {

	public AdminAuthentication(
		Object principal,
		Object credentials,
		Collection<? extends GrantedAuthority> authorities
	) {
		super(principal, credentials, authorities);
	}

	public Long getAdminId() {
		return Long.valueOf(getPrincipal().toString());
	}
}

File: support/security-web/src/main/kotlin/com/beat/support/security/authentication/internal/AdminAuthentication.kt
package com.beat.support.security.authentication.internal

import org.springframework.security.authentication.UsernamePasswordAuthenticationToken
import org.springframework.security.core.GrantedAuthority

/** `ROLE_ADMIN` 인증 토큰. principal은 항상 관리자 memberId다. */
internal class AdminAuthentication(
    memberId: Long,
    authorities: Collection<GrantedAuthority>,
) : UsernamePasswordAuthenticationToken(memberId, null, authorities)
